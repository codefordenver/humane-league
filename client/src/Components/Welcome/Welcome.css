.Welcome {
  display: grid;
  grid-template-columns: 50% 50%;
  height: 100%;
  align-items: center;
  justify-content: space-around;
  width: 100%; }
  .Welcome a {
    height: 70px;
    display: flex;
    width: 250px;
    align-items: center;
    justify-content: center;
    border-radius: 2px;
    border: none;
    background: #ffffffe3;
    cursor: pointer;
    margin: 0px auto;
    transition: 0.3s;
    font-size: 20px;
    padding: 10px; }
    .Welcome a:hover {
      background: white; }
  .Welcome .fan-logo {
    width: 100%; }
  .Welcome .welcome-controls {
    color: #fff;
    font-size: 20pt;
    text-shadow: 2px 2px 15px rgba(0, 0, 0, 0.6);
    display: grid;
    position: relative;
    grid-template-columns: 1fr;
    grid-gap: 20px;
    padding: 0px 80px;
    justify-items: center; }
    .Welcome .welcome-controls button {
      height: 70px;
      display: flex;
      width: 250px;
      align-items: center;
      justify-content: center;
      border-radius: 2px;
      box-shadow: 0px 0px 5px rgba(0, 0, 0, 0.6);
      border: none;
      background: #ffffffe3;
      color: #000;
      cursor: pointer;
      transition: 0.3s;
      font-size: 20px;
      padding: 10px; }
      .Welcome .welcome-controls button.primary {
        z-index: 5; }
      .Welcome .welcome-controls button.none {
        display: none; }
      .Welcome .welcome-controls button.hidden {
        position: absolute;
        background: transparent;
        color: transparent;
        width: 150px;
        margin-left: 90px;
        overflow: hidden; }
      .Welcome .welcome-controls button:hover {
        background: #ffffff; }
  .Welcome .create-account-form {
    flex-direction: column;
    align-items: center;
    width: 500px; }
    .Welcome .create-account-form .create-account-user-input,
    .Welcome .create-account-form .create-account-submit {
      border: none;
      box-shadow: 2px 2px 15px rgba(0, 0, 0, 0.6);
      font-size: 20pt;
      margin: 20px;
      padding: 20px; }
    .Welcome .create-account-form .create-account-submit {
      cursor: pointer;
      background: rgba(0, 0, 0, 0.6);
      color: #fff; }
      .Welcome .create-account-form .create-account-submit:hover {
        background: #046A38; }

.welcome-description {
  font-size: 20px;
  line-height: 1.5em;
  margin: 30px;
  padding: 20px;
  background-color: #ffffffe3;
  border-radius: 2px;
  box-shadow: 2px 2px 15px rgba(0, 0, 0, 0.6); }

@media screen and (max-width: 1150px) {
  .Welcome {
    grid-template-columns: 90%;
    grid-template-rows: 50% 50%;
    padding-bottom: 50px; }
    .Welcome .fan-logo {
      margin: 0 auto;
      width: 400px; }
    .Welcome .welcome-controls {
      padding: 0; } }

@media screen and (max-width: 550px) {
  .Welcome .fan-logo {
    width: 90%; } }
