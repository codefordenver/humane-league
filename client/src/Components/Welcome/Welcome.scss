.Welcome {
  display: grid;
  grid-template-columns: 50% 50%;
  height: 100%;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  .fan-logo {
    // margin: 0 auto;
    // display: block;
    // transition: margin 1s ease;
    width: 100%;
  }
  .fan-logo.shifted {
    // margin-right: 0;
    // transition: margin 1s ease;
    align-self: flex-end;
  }
  .welcome-controls {
    color: #fff;
    font-size: 20pt;
    text-shadow: 2px 2px 15px rgba(0, 0, 0, 0.6);
    display: grid;
    position: relative;
    grid-template-columns: 1fr;
    grid-gap: 20px;
    padding: 0px 80px;
    justify-items: center;
    button {
      height: 70px;
      display: flex;
      width: 250px;
      align-items: center;
      justify-content: center;
      border-radius: 2px;
      border: none;
      background: #1CB841;
      color: white;
      cursor: pointer;
      transition: 0.3s;
      font-size: 20px;
      padding: 10px;
      &.primary {
        z-index: 5;
      }
      &.none {
        display: none;
      }
      &.hidden {
        position: absolute;
        width: 150px;
        margin-left: 90px;
        overflow: hidden;
      }
      &:hover {
        color: white;
        background-color: rgb(50, 50, 50);
      }
    }
  }
  .create-account-form {
    display: flex;
    flex-direction: column;
    width: 500px;
    .create-account-user-input,
    .create-account-submit {
      border: none;
      box-shadow: 2px 2px 15px rgba(0, 0, 0, 0.6);
      font-size: 20pt;
      margin: 20px;
      padding: 20px;
    }
    .create-account-submit {
      cursor: pointer;
      background: rgba(0, 0, 0, 0.6);
      color: #fff;
      &:hover {
        background: #046A38;
      }
    }
  }
}

.Welcome.shifted {
  flex-direction: row;
}

.welcome-description {
  background-color: white;
  font-size: 20px;
  line-height: 1.5em;
  margin: 30px;
  padding: 20px;
}
