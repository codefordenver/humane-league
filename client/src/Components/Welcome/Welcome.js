import React, { Component } from 'react';
import fanLogo from '../../assets/fan-logo.png';
import './Welcome.css';
import firebase from '../../firebase';


class Welcome extends Component {
  constructor() {
    super();
    this.state = {
      showForm: false,
    }
  }

  handleClick = () => {
    this.setState({ showForm: true });
    const provider = new firebase.auth.GoogleAuthProvider();

    firebase.auth().signInWithPopup(provider).then(function(result) {
      // This gives you a Google Access Token. You can use it to access the Google API.
      var token = result.credential.accessToken;
      // The signed-in user info.
      console.log('token:', token)
      var user = result.user;
      // ...
      console.log('user:', user)

    }).catch(function(error) {
      // Handle Errors here.
      var errorCode = error.code;
      var errorMessage = error.message;
      // The email of the user's account used.
      var email = error.email;
      // The firebase.auth.AuthCredential type that was used.
      var credential = error.credential;
      // ...
    });
  }

  render() {
    const signInPrompt = 
      <h1 className='welcome-controls'>
        <span onClick={this.handleClick}>create account</span>/
        <span onClick={this.handleClick}>sign-in</span>
      </h1>;

    const accountForm = 
      <form className='create-account-form'>
        <input className='create-account-user-input' type='text' placeholder='user-name' />
        <button className='create-account-submit' >Create Account</button>
      </form>;

    const imgClassName = !this.state.showForm ? 'fan-logo' : 'fan-logo shifted';
    const welcomeClassName = !this.state.showForm ? 'Welcome' : 'Welcome shifted';

    const controls = !this.state.showForm ? signInPrompt : accountForm;

    return (
      <div className={welcomeClassName}>
        <img className={imgClassName} src={fanLogo} alt='' />
        {controls}
      </div>
    )
  }
}

export default Welcome;