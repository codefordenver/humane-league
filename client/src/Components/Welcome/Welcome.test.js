import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Welcome } from './Welcome';

describe('Welcome component tests', () => {
  let renderedWelcome;
  let defaultState;
  
  beforeEach(() => {
    renderedWelcome = shallow( <Welcome User={{ name: 'Julie Hawkins' }}/> );
    defaultState = {
      showForm: false,
      errorMessage: null
    }
  })
  it('renders without crashing', () => {
    expect(renderedWelcome).toBeDefined();
    expect(renderedWelcome.state()).toEqual(defaultState);
  });
  it('matches SnapShot', () => {
    expect(renderedWelcome).toMatchSnapshot();
  })
})