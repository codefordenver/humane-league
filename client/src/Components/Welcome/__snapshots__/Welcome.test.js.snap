// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome component tests matches SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    User={
        Object {
            "name": "Julie Hawkins",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          className="fan-logo shifted"
          src="fan-logo.png"
/>,
        <div>
          <p
                    className="welcome-description"
          >
                    Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!
          </p>
          <Link
                    replace={false}
                    to="/home"
          >
                    Dashboard
          </Link>
</div>,
      ],
      "className": "Welcome shifted",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "className": "fan-logo shifted",
          "src": "fan-logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="welcome-description"
>
              Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!
</p>,
            <Link
              replace={false}
              to="/home"
>
              Dashboard
</Link>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!",
              "className": "welcome-description",
            },
            "ref": null,
            "rendered": "Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Dashboard",
              "replace": false,
              "to": "/home",
            },
            "ref": null,
            "rendered": "Dashboard",
            "type": [Function],
          },
          null,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            className="fan-logo shifted"
            src="fan-logo.png"
/>,
          <div>
            <p
                        className="welcome-description"
            >
                        Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!
            </p>
            <Link
                        replace={false}
                        to="/home"
            >
                        Dashboard
            </Link>
</div>,
        ],
        "className": "Welcome shifted",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "className": "fan-logo shifted",
            "src": "fan-logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="welcome-description"
>
                Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!
</p>,
              <Link
                replace={false}
                to="/home"
>
                Dashboard
</Link>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!",
                "className": "welcome-description",
              },
              "ref": null,
              "rendered": "Welcome to The Humane League's Fast Action Network! FAN gives individuals a way to be heard by the largest corporations in the world. Whether it is by signing a petition, sending a tweet, emailing a CEO, or posting a quick comment on Facebook, this team comes together for a few minutes each week to call on companies to end their support of the worst factory farm cruelties, and it works! When activists like yourself come together for animals, the opportunities are endless!",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Dashboard",
                "replace": false,
                "to": "/home",
              },
              "ref": null,
              "rendered": "Dashboard",
              "type": [Function],
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
